### DISCUSSION TITLE
MG_IO_SIZE

### DISCUSSION THREAD

**Discussion by lonejack** [cesanta/mongoose#2736] [Ask questions, get answers] [answered]
Hi, 
I'm running mongoose for a mqtt test on linux as client.
The version used is 7.13. 52997c6c5170050148d23436d2ed398811f48c30
For this implementation I started from:
examples/mqtt-client

The problem faced.
When I exchange messages with broker I'm allowed to see messages where the length in term of size is less than MG_IO_SIZE.
I I want to receive messages much larger I've to modify the MG_IO_SIZE parameter.

On the documentation I see the meaning of the parameter:
**MG_IO_SIZE	Granularity of the send/recv IO buffer growth**

To me seems that if you have to receive a greater message the mongoose provide to enlarge automatically the buffer.
Am I right?
Currently I changed the parameter from 2048 to 16384 and it works...

**Comment by scaprile** [comment, marked as answer]
Mongoose resizes its buffer in MG_IO_SIZE-size chunks up to MG_MAX_RECV_SIZE.
https://github.com/cesanta/mongoose/blob/497df68de11a1a326f7061e8d588517ed264688d/mongoose.c#L7386-L7397
It is very hard for me to guess what is going on with such a description. What _exactly_ happens ? 
Please sniff your network (using Wireshark, for example) and observe the traffic.
Please see the logs.
What fails ? When ? Where ?
If, after your network inspection, you find an actual issue, feel free to open an issue, fill the issue template, and provide us with pertinent information (including useful network captures) in order to reproduce and fix the issue. We don't inspect your code for errors, please send the minimum piece of code that can reproduce the issue
